src/API.jl

```diff
@@ -0,0 +1,56 @@
+module API
+
+import Pkg3
+using Pkg3.Types
+using Base.Random.UUID
+
+add(pkg::String)               = add([pkg])
+add(pkgs::Vector{String})      = add([PackageSpec(pkg) for pkg in pkgs])
+add(pkgs::Vector{PackageSpec}) = add(EnvCache(), pkgs)
```

Comment:
Perhaps `add(pkgs::Vector{<:Union{String,PackageSpec}})`? Ditto with the others. Of course, this is a stricter API so we could always loosen to that if it turns out to be useful.